src/FakeItEasy.Analyzer/SymbolExtensions.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace FakeItEasy.Analyzer
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.CodeAnalysis;
+
+    public static class SymbolExtensions
+    {
+        public static string GetFullName(this INamedTypeSymbol type)
+        {
+            if (type == null)
+            {
+                throw new ArgumentNullException(nameof(type));
+            }
+
+            var stack = new Stack<string>();
+            stack.Push(type.Name);
+            var tmp = type.ContainingType;
+            while (tmp != null)
+            {
+                type = tmp;
+                stack.Push(tmp.Name);
+                tmp = tmp.ContainingType;
+            }
+
+            var ns = type.ContainingNamespace;
+            while (!string.IsNullOrEmpty(ns?.Name))
+            {
+                stack.Push(ns.Name);
+                ns = ns.ContainingNamespace;
+            }
+
+            return string.Join(".", stack);
+        }
+    }
+}

```

Comment:
I'm sure this code was template code, but the variable names leave something to be desired.